gp-nested-forms/gpnf-override-parent-merge-tag-on-submission.php

```diff
@@ -7,66 +7,115 @@
  *
  * Override all {Parent} merge tags when the parent form is submitted or a parent entry is updated.
  */
-function gpnf_override_parent_merge_tags( $entry, $form ) {
-	// Update '123' to the ID of the Child form. Set as false to apply to all child forms.
-	$id_of_child_form = 123;
-	// Updated '4, 5, 6' with the IDs of the fields you want to skip. Leave blank to apply to all fields.
-	$exclude_field_ids = array( 4, 5, 6 );
-
-	foreach ( $form['fields'] as $field ) {
-		if ( $field->get_input_type() === 'form' ) {
-			$child_form_id = $field->gpnfForm;
-			if ( $id_of_child_form ) {
-				if ( $child_form_id != $id_of_child_form ) {
-					return $entry;
-				}
+
+class GPNF_Override_Parent_Merge_Tags {
+
+	/**
+	 * @var int|null child form ID to apply the override to. If null, override will apply to all child forms.
+	 */
+	private $child_form_id = null;
+
+	/**
+	 * @var int[] array field ID's to exclude when overriding a child form's {Parent} merge tags.
+	 */
+	private $exclude_field_ids = array();
+
+	function __construct( $config = null ) {
+		$this->set_config_data( $config );
+		$this->init_hooks();
+	}
+
+	function set_config_data( $config ) {
+		if ( empty( $config ) ) {
+			return;
+		}
+
+		$this->child_form_id = $config['child_form_id'];
+
+		if ( array_key_exists( 'exclude_field_ids', $config ) ) {
```

Comment:
Unfortunately `rgar` might not be found by the interpreter at this point since this could be run before gravity forms is.

Reply:
I dig! I think in another PR I mentioned that we typically go with `isset()` or `! empty()` for brevity. I'm not gonna hold the PR up over this though.

Reply:
Yup, I believe so. I wrote this the same day (before your other review) and didn't even think about updating here so thanks for the reminder!

Reply:
Yup, I believe so. I wrote this the same day (before your other review) and didn't even think about updating here so thanks for the reminder!